tbx/static_src/javascript/components/mobile-menu.js

```diff
@@ -19,6 +19,28 @@ class MobileMenu {
         this.node.addEventListener('click', () => {
             this.toggle();
         });
+
+        // Close navigation if we focus anywhere outside the menu
+        document.addEventListener('focusin', (e) => {
+            const inMenu = !!e.target.closest('[data-mobile-menu]');
+
+            if (!inMenu) {
+                if (this.state.open) {
+                    this.close();
+                    this.state.open = false;
+                }
+            }
+        });
+
+        // Close mobile dropdown with escape key for improved accessibility
```

Comment:
Any particular reason not to do this on desktop as well?

Reply:
Good point - I will add that.

Reply:
Added in `desktop-close-menus.js`